packages/unit/src/main.ts

```diff
@@ -23,45 +24,58 @@ const SuitesDIAdapters = {
+): { create: <TBuilder>(type: Type<TestBedBuilder<TClass>>) => TBuilder } {
+  return {
+    create: <TBuilder>(type: Type<TestBedBuilder<TClass>>): TBuilder => {
+      const diPackageResolver = new PackageResolver<DependencyInjectionAdapter>(diAdapters);
 
-  const diAdapter = diPackageResolver
-    .resolveCorrespondingAdapter()
-    .then((adapter) => adapter)
-    .catch(() => {
-      throw new AdapterNotFoundError(`Suites requires an adapter to integrate with different dependency injection frameworks.
+      const diAdapter = diPackageResolver
+        .resolveCorrespondingAdapter()
+        .then((adapter) => adapter)
+        .catch(() => {
+          throw new AdapterNotFoundError(`Suites requires an adapter to integrate with different dependency injection frameworks.
 It seems that you haven't installed an appropriate package. To resolve this issue, please install
 one of the available packages that matches your dependency injection framework.
 Refer to the docs for further information: https://suites.dev/docs`);
-    });
+        });
 
-  const doublesPackageResolver = new PackageResolver<MockFunction<unknown>>(doublesAdapters);
+      const doublesPackageResolver = new PackageResolver<MockFunction<unknown>>(doublesAdapters);
 
-  const doublesAdapter = doublesPackageResolver
-    .resolveCorrespondingAdapter()
-    .then((adapter) => adapter)
-    .catch(() => {
-      throw new AdapterNotFoundError(`Suites requires an adapter to integrate with different mocking libraries.
+      const doublesAdapter = doublesPackageResolver
+        .resolveCorrespondingAdapter()
+        .then((adapter) => adapter)
+        .catch(() => {
+          throw new AdapterNotFoundError(`Suites requires an adapter to integrate with different mocking libraries.
 It seems that you haven't installed an appropriate adapter package. To resolve this issue, please install
 one of the available packages that matches your mocking library.
 Refer to the docs for further information: https://suites.dev/docs`);
-    });
+        });
 
-  const unitMocker = new UnitMocker(doublesAdapter);
-  return new TestBedBuilder<TClass>(doublesAdapter, diAdapter, unitMocker, targetClass, console);
+      const unitMocker = new UnitMocker(doublesAdapter, diAdapter);
+
+      return new type(doublesAdapter, diAdapter, unitMocker, targetClass, console) as TBuilder;
+    },
+  };
 }
 
 export class TestBed {
-  public static solitary<TClass = any>(targetClass: Type<TClass>): TestBedBuilder<TClass> {
-    return createTestbedBuilder(SuitesDIAdapters, SuitesDoublesAdapters, targetClass);
+  public static solitary<TClass = any>(targetClass: Type<TClass>): SolitaryTestBedBuilder<TClass> {
+    return testBedBuilderFactory(SuitesDIAdapters, SuitesDoublesAdapters, targetClass).create(
+      SolitaryTestBedBuilder
+    );
+  }
+
+  public static sociable<TClass = any>(targetClass: Type<TClass>): SociableTestBedBuilder<TClass> {
```

Comment:
I couldn't find any tests covering this new user-facing API?
I see tests that cover the insides, but nothing drives it as a user would?

Reply:
There is `e2e` that I need to apply as well, you are right